UNO/Database/Database.java

```diff
@@ -0,0 +1,160 @@
+package Database;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.sql.*;
+import java.util.Properties;
+
+public class Database {
+
+    private Connection connection;
+    private String key;
+
+    public Database(String property_file_name)
+    {
+
+        // Declare local variables.
+        Properties properties = new Properties();
+        FileReader reader = null;
+        String url, username, password;
+
+        // Attempt to read the properties file.
+        try
+        {
+
+            // Read the contents of the file into the file reader.
+            reader = new FileReader(property_file_name);
+
+            // Load the data from the file into the 'properties' object.
+            properties.load(reader);
+
+            // Close the connection to the file.
+            reader.close();
+
+        }
+        catch (IOException exception)
+        {
+
+            // Display the exception information to the user.
+            exception.printStackTrace();
+
+            // Terminate the program with an error code.
+            System.exit(-1);
+
+        }
+
+        // Obtain the url, username, password, and key from the properties file.
+        url = properties.getProperty("url");
+        username = properties.getProperty("user");
+        password = properties.getProperty("password");
+        key = properties.getProperty("key");
+
+        // Attempt to connect to the database.
+        try
+        {
+
+            // Connect to the database.
+            connection = DriverManager.getConnection(url, username, password);
+
+        }
+        catch (SQLException exception)
+        {
+
+            // Display the exception information to the user.
+            exception.printStackTrace();
+
+            // Terminate the program with an error code.
+            System.exit(-1);
+
+        }
+
+    }
+    
+    public Connection getConnection()
+    {
+    	return connection;
+    }
+
+    public boolean verifyAccount(LoginData data)
```

Comment:
Possible suggestion that could be wrong, but instead of constructing SQL queries by concatenating strings, use prepared statements. 
```suggestion
public boolean verifyAccount(LoginData data) {
    // Declare local variables.
    PreparedStatement statement = null;
    ResultSet results = null;
    boolean flag = false;
    
    try {
        // Prepare the statement with parameterized query.
        statement = connection.prepareStatement("SELECT COUNT(*) FROM uno_user WHERE username=? AND password=?");
        statement.setString(1, data.getUsername());
        statement.setString(2, data.getPassword());

        // Execute the query.
        results = statement.executeQuery();

        // Seek to the first record.
        results.next();

        // Determine if the login data matched any account.
        flag = (results.getInt(1) > 0);
    } catch (SQLException exception) {
        // Handle exceptions
        exception.printStackTrace();
        return false;
    } finally {
        // Close resources
        try {
            if (results != null) results.close();
            if (statement != null) statement.close();
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
    return flag;
}
```

Reply:
Do the prepared statements prevent from SQL injection attacks?

Reply:
Yes. Using prepared statements with parameterized queries helps prevent SQL injection attacks by separating SQL code from user input.

Reply:
I think that would be a great change.